ui/src/app/views/recent-work/component.ts

```diff
@@ -14,21 +23,57 @@ import {
 export class RecentWorkComponent implements OnInit {
   resourceList: RecentResource[];
   fullList: RecentResource[] = [];
+  @Input('workspace')
+  workspace: Workspace;
```

Comment:
There should probably be a test case that hits this new workspace-input codepath.